editoast/src/views/rolling_stock/towed.rs

```diff
@@ -0,0 +1,610 @@
+    /// New locked value
+    pub locked: bool,
+}
+
+#[utoipa::path(
+    patch, path = "",
+    tag = "rolling_stock",
+    params(TowedRollingStockIdParam),
+    request_body = TowedRollingStockLockedForm,
+    responses(
+        (status = 204, description = "No content when successful")
+    )
+)]
+async fn patch_by_id_locked(
+    State(db_pool): State<DbConnectionPoolV2>,
+    Extension(authorizer): AuthorizerExt,
+    Path(towed_rolling_stock_id): Path<i64>,
+    Json(TowedRollingStockLockedForm { locked }): Json<TowedRollingStockLockedForm>,
+) -> Result<StatusCode> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::RollingStockCollectionWrite].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    // FIXME: check that the towed rolling stock exists (the Option<RollingStockModel> is ignored here)
+    // should return a 404 instead of a 204 in case the identifier doesn't exist.
+    TowedRollingStockModel::changeset()
+        .locked(locked)
+        .update(conn, towed_rolling_stock_id)
+        .await?;
+
+    Ok(StatusCode::NO_CONTENT)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::TowedRollingStock;
+    use super::TowedRollingStockCountList;
+    use crate::views::test_app::TestAppBuilder;
+    use axum::http::StatusCode;
+    use rstest::rstest;
+    use serde_json::json;
+    use uuid::Uuid;
+
+    #[rstest]
+    async fn create_and_list_towed_rolling_stock() {
+        let app = TestAppBuilder::default_app();
+
+        let name = Uuid::new_v4().to_string();
+        let towed_rolling_stock_json = json!({
+            "name": name,
+            "locked": false,
+            "mass": 42000,
+            "length": 16500,
+            "comfort_acceleration": 0.05,
+            "startup_acceleration": 0.05,
+            "inertia_coefficient": 1.0,
+            "rolling_resistance": {
+                "type": "yoda",
+                "A": 1000.0,
+                "B": 100.0,
+                "C": 10.0,
+            },
+            "gamma": {
+                "type": "CONST",
+                "value": 1.0,
+            },
+        });
+
+        let request = app
+            .post("/towed_rolling_stock")
+            .json(&towed_rolling_stock_json);
+
+        let towed_rolling_stock: TowedRollingStock =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+        assert_eq!(towed_rolling_stock.name, name);
+
+        let towed_rolling_stocks: TowedRollingStockCountList = app
+            .fetch(app.get("/towed_rolling_stock"))
+            .assert_status(StatusCode::OK)
+            .json_into();
+
+        assert!(!towed_rolling_stocks.results.is_empty());
+    }
+
+    #[rstest]
+    async fn get_unknown_towed_rolling_stock() {
+        let app = TestAppBuilder::default_app();
+
+        let id: i64 = rand::random();
```

Comment:
Not a huge fan of unbounded randomness in tests. What about
```suggestion
        let id: i64 = i64::MAX;
```

Because a failure can be hard to diagnose if `id` occasionally designates a towed rolling stock that already exists in the DB.

Reply:
Mmmm, I'd say we should be pretty safe. A lot of systems actually use UUID as identifier in production (not in tests), assuming there will never be a collision (the chance is so low). From what I understand in [this documentation](https://en.wikipedia.org/w/index.php?title=Universally_unique_identifier&oldid=755882275#Random_UUID_probability_of_duplicates), we should be able to calculate the probability of collision for an `i64` after generating a million of rolling stocks (which is already way more than we will ever have).

$`\frac{{1000000}^2}{2 • 2^{64}} = 2.710 • 10^{-8}`$

I agree `i64` is still way less than an UUID (which is 128 bits), but still, I'd say it's OK.

Also, I don't see `i64::MAX` as a good solution either because what if I want to write another test? In another file? It's going to start to be unmaintainable to find identifiers.

What do you think?

Reply:
Sure, you're right. I just got a bit triggered by having randomness in a non property-based test, but factually the probability of things going wrong is indeed very low.

> Also, I don't see i64::MAX as a good solution either because what if I want to write another test?

Our tests are isolated in their own transaction so there shouldn't be conflicting IDs coming from the tests. ID conflicts are still possible with data that's already there from before the execution of the test though. 

Resolving this (especially since we have bigger flakiness issues on our hands already 🙃).